openquake/baselib/general.py

```diff
@@ -552,6 +567,14 @@ def __mul__(self, other):
     def __truediv__(self, other):
         return self * (1. / other)
 
+    def __missing__(self, key):
+        if self.accum is None:
+            # no accumulator, accessing a missing key is an error
+            raise KeyError(key)
+        val = self[key] = copy.deepcopy(self.accum)
```

Comment:
Please add a doctest, a test or a comment describing a use case in which deepcopy is useful